components/kyma-environment-broker/internal/storage/dbsession/read.go

```diff
@@ -373,24 +373,28 @@ func (r readSession) GetNumberOfInstancesForGlobalAccountID(globalAccountID stri
 	return res.Total, err
 }
 
-func (r readSession) ListInstances(pageSize int, page int) ([]internal.Instance, int, int, error) {
+func (r readSession) ListInstances(filter dbmodel.InstanceFilter) ([]internal.Instance, int, int, error) {
 	var instances []internal.Instance
 
-	order, err := pagination.ConvertPageSizeAndOrderedColumnToSQL(pageSize, page, postsql.CreatedAtField)
-	if err != nil {
-		return nil, -1, -1, errors.Wrap(err, "while converting page and pageSize to SQL statement")
-	}
+	// Base select and order by created at
+	stmt := r.session.
+		Select("*").
+		From(postsql.InstancesTableName).
+		OrderBy(postsql.CreatedAtField)
 
-	stmtWithPagination := fmt.Sprintf("SELECT * FROM %s %s", postsql.InstancesTableName, order)
+	// Add pagination
+	if filter.Page > 0 && filter.PageSize > 0 {
```

Comment:
Can such a case that Page is 0 or PageSize is 0 occur? I thought that there was validation for that

Reply:
Actually I think not, because the handler code does the validation already. I added a check also here when a different part of code will use that in the future and forgets to validate pagination params. 
Do you think I should remove this check ?

Reply:
Ok, I think that one more If won't hurt :D